ra2ce/analysis/analysis_collection.py

```diff
@@ -46,11 +47,11 @@ def from_config(cls, analysis_config: AnalysisConfigWrapper) -> AnalysisCollecti
         """
         return cls(
             direct_analyses=[
-                AnalysisFactory(analysis).get_analysis(analysis_config)
+                AnalysisFactory(analysis).get_direct_analysis(analysis_config)
```

Comment:
If you are going to use the same factory, as an object, you are better off defining it earlier on, then reusing it for each different method. Otherwise you should make both methods static and let them consume the analysis type (`analysis.analysis`)
```suggestion
            _factory = AnalysisFactory(analysis)
                ...
                _factory.get_direct_analysis(analysis_config)
```
or
```suggestion
                AnalysisFactory.get_direct_analysis(analysis, analysis_config)
```

Reply:
I don't understand the first suggestion, as the factory is created for each different separate analysis.
The second seems valid, although previously we said we don't like static methods.

Reply:
You are right, I read this incorrectly, however it begs the question whether the factory could be slightly improved by being initialized with an `analysis_config` and then getting the corresponding analysis based on the input of this class method.

Reply:
Next steps:
* Create new classes for direct and indirect